### PR TITLE
Add support for Gtest as a test protocol

### PR DESCRIPTION
**Pull request by dcbaker**
GTest is a pretty common format in the C/C++ world. It's also a pretty powerful test framework, able to give more detailed results than you get from just the exitcode. In particular it can output JUnit xml. Since we now have support for JUnit xml generation ourselves we can inject such arguments, then consume the resulting JUnit and put it in our own XML.

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -3823,8 +3823,8 @@ def add_test(self, node, args, kwargs, is_base_test):
         if not isinstance(timeout, int):
             raise InterpreterException('Timeout must be an integer.')
         protocol = kwargs.get('protocol', 'exitcode')
-        if protocol not in ('exitcode', 'tap'):
-            raise InterpreterException('Protocol must be "exitcode" or "tap".')
+        if protocol not in {'exitcode', 'tap', 'gtest'}:
+            raise InterpreterException('Protocol must be "exitcode", "tap", or "gtest".')
```
Comment (jon-turney) [inline mesonbuild/interpreter.py:3827]: Does this need a FeatureNew call when protocol  == gtest?

File: docs/markdown/Reference-manual.md [hunk 2]
```diff
@@ -1664,9 +1664,10 @@ test(..., env: nomalloc, ...)
 
 - `protocol` specifies how the test results are parsed and can be one
   of `exitcode` (the executable's exit code is used by the test harness
-  to record the outcome of the test) or `tap` ([Test Anything
-  Protocol](https://www.testanything.org/)). For more on the Meson test
-  harness protocol read [Unit Tests](Unit-tests.md). Since 0.50.0
+  to record the outcome of the test), `tap` ([Test Anything
+  Protocol](https://www.testanything.org/)) or `gtest`. For more on the Meson
+  test harness protocol read [Unit Tests](Unit-tests.md). Since 0.50.0.
+  *(Since 0.55.0)* The `gtest` protocol.
```
Comment (jon-turney) [inline docs/markdown/Reference-manual.md:1670]: This might be best written as a bulleted list.

Comment (dcbaker) [reply docs/markdown/Reference-manual.md]: I agree, I've updated it to be a list

File: mesonbuild/interpreter.py (hunk 1 above)
Comment (dcbaker) [reply mesonbuild/interpreter.py:3829]: Yup, I definitely missed that.

**Comment by dcbaker** [comment]
Updated to address @jon-turney's comments. I've also updated the Unit-test.md documentation.

File: docs/markdown/Reference-manual.md [hunk 3]
```diff
@@ -1663,10 +1663,13 @@ test(..., env: nomalloc, ...)
   Since 0.46.0
 
 - `protocol` specifies how the test results are parsed and can be one
-  of `exitcode` (the executable's exit code is used by the test harness
-  to record the outcome of the test) or `tap` ([Test Anything
-  Protocol](https://www.testanything.org/)). For more on the Meson test
-  harness protocol read [Unit Tests](Unit-tests.md). Since 0.50.0
+  of `exitcode`, `tap`, or `gtest. For more informatoin about test harness
```
Comment (jpakkane) [inline docs/markdown/Reference-manual.md:1666]: Missing backtick after `gtest`.

Comment (QuLogic) [reply docs/markdown/Reference-manual.md:1666]: type: informatoin

Comment (dcbaker) [reply docs/markdown/Reference-manual.md]: Both issues fixed.

**Comment by dcbaker** [comment]
I think the macos failure is suprious

**Comment by jpakkane** [comment]
Looks fine by me. Anyone else have comments?

**Pull request merged**